### PR TITLE
Admin user roles

### PR DESCRIPTION
**Pull request by bak45247**

### CODE REVIEW

File: tinker/admin/user_roles/__init__.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            current_username = session['admin_viewer_username']
+        else:
+            current_username = session['username']
+
+        # get username
+        username = request.form.get('username')
+
+        if username:
+            session.clear()
+            # session_keys = copy.deepcopy(session.keys())
```
Comment (c-schwarze) [inline tinker/admin/user_roles/__init__.py]: Remove some commented out code?

File: tinker/admin/user_roles/__init__.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            #     session.pop(key)
+
+            session['admin_username'] = current_username
+            session['username'] = username
+        else:
+            return 'error'
+
+        session['admin_viewer'] = True
+
+        # session.modified = True
```
Comment (c-schwarze) [inline tinker/admin/user_roles/__init__.py]: Some more commented out code to remove

File: tinker/tinker_controller.py [hunk 3]
```diff
@@ -130,7 +130,7 @@ def init_user():
                 day_is_passed = True
 
             # if not production, then clear our session variables on each call
-            if dev or day_is_passed:
+            if (not session.get('admin_viewer', False)) and (dev or day_is_passed):
```
Comment (c-schwarze) [inline tinker/tinker_controller.py]: Did you add in the code for the "day_is_passed" functionality?

File: tinker/views.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
 from flask_classy import FlaskView
-from flask import Flask, Blueprint
+from flask import Flask, Blueprint, session
```
Comment (c-schwarze) [inline tinker/views.py]: You added session, but it doesn't look like you used it?

File: tinker/tinker_controller.py (hunk 3 above)
Comment (bak45247) [inline tinker/tinker_controller.py]: no, that logic was already there. I just added the part to check if we were viewing with admin_viewer == True

File: tinker/admin/user_roles/__init__.py (hunk 1 above)
Comment (bak45247) [inline tinker/admin/user_roles/__init__.py]: deleted

File: tinker/admin/user_roles/__init__.py (hunk 2 above)
Comment (bak45247) [inline tinker/admin/user_roles/__init__.py]: deleted

File: tinker/views.py (hunk 4 above)
Comment (bak45247) [inline tinker/views.py]: fixed, I added that before and i just forgot to delete it

File: tinker/tinker_controller.py (hunk 3 above)
Comment (c-schwarze) [inline tinker/tinker_controller.py]: oh gotcha. Wasn't looking carefully enough. 👍

**Pull request merged**